tests/unit/Firestore/TransactionTest.php

```diff
@@ -201,6 +201,127 @@ public function testDelete()
         ]);
     }
 
+    /**
+     * @dataProvider documents
+     */
+    public function testDocuments(array $input, array $names)
+    {
+        $name = function ($name) {
+            return sprintf('projects/%s/databases/%s/documents/%s', self::PROJECT, self::DATABASE, $name);
+        };
+
+        $res = [
+            [
+                'found' => [
+                    'name' => $names[0],
+                    'fields' => [
+                        'hello' => [
+                            'stringValue' => 'world'
+                        ]
+                    ]
+                ],
+                'readTime' => ['seconds' => 1, 'nanos' => 0]
+            ], [
+                'missing' => $names[1],
+                'readTime' => ['seconds' => 1, 'nanos' => 0]
+            ], [
+                'missing' => $names[2],
+                'readTime' => ['seconds' => 1, 'nanos' => 0]
+            ]
+        ];
+
+        $this->connection->batchGetDocuments(Argument::allOf(
+            Argument::withEntry('documents', $names),
+            Argument::withEntry('transaction', self::TRANSACTION)
+        ))->shouldBeCalled()->willReturn($res);
+
+        $this->transaction->___setProperty('connection', $this->connection->reveal());
+
+        $res = $this->transaction->documents($input);
+
+        $this->assertEquals('world', $res[0]['hello']);
+        $this->assertCount(3, $res);
+    }
+
+    public function documents()
+    {
+        $b = $this->prophesize(DocumentReference::class);
+        $b->name()->willReturn('a/b');
+
+        $c = $this->prophesize(DocumentReference::class);
+        $c->name()->willReturn('a/c');
+
+        $d = $this->prophesize(DocumentReference::class);
+        $d->name()->willReturn('a/d');
+        return [
+            [
+                [
+                    'a/b',
+                    'a/c',
+                    'a/d'
+                ], [
+                    'projects/'. self::PROJECT .'/databases/'. self::DATABASE .'/documents/a/b',
```

Comment:
Could this just reference a constant? (same with other applications through the test)

e.g.

```php
sprintf(self::TEMPLATE, self::PROJECT, self::DATABASE, 'b')
```